validator/server_api/json.go

```diff
@@ -103,3 +41,51 @@ func MachineStepResultFromJson(resultJson *MachineStepResultJson) (*validator.Ma
 		GlobalState: resultJson.GlobalState,
 	}, nil
 }
+
+func RedisStreamForRoot(moduleRoot common.Hash) string {
+	return fmt.Sprintf("stream:%s", moduleRoot.Hex())
+}
+
+type Request struct {
+	Input      *InputJSON
+	ModuleRoot common.Hash
+}
+
+type InputJSON struct {
+	Id            uint64
+	HasDelayedMsg bool
+	DelayedMsgNr  uint64
+	PreimagesB64  map[arbutil.PreimageType]*jsonapi.PreimagesMapJson
+	BatchInfo     []BatchInfoJson
+	DelayedMsgB64 string
+	StartState    validator.GoGlobalState
+}
+
+type BatchInfoJson struct {
+	Number  uint64
+	DataB64 string
+}
+
+type RedisValidationServerConfig struct {
```

Comment:
the config , defaults and addoptions belongs in valnode/redisconsumer.go

Reply:
We can't just move them there though, it will introduce cyclic dependency valnode->server_arb ->valnode. 

validator_spawner.go (server_arb) initializes redis validation server config (calls ValidationServerConfigAddOptions, that would be in valnode if I moved).
valnode.go itself spawns validator spawners so has dependency on server_arb.


I have moved out redis consumer (and also producer for consistency) into separate packages to accomodate your comment.